File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        int complement;
        Map<Integer, Integer> numIx = new HashMap<>();

        for (int i=0; i < n; i++) {
            complement = target - nums[i];
            if (numIx.containsKey(complement)) {
                return new int[]{numIx.get(complement), i};
            } else {
                numIx.put(nums[i], i);
            }
        }

        return new int[]{};
    }
}

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        index = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in index:
                return [i, index[complement]]
            index[num] = i
        return [0,0]

        